internal/beater/monitoringtest/opentelemetry.go

```diff
@@ -56,49 +106,39 @@ func assertOtelMetrics(t *testing.T, reader sdkmetric.Reader, expectedMetrics ma
 			case metricdata.Gauge[int64]:
 				assert.Equal(t, 1, len(d.DataPoints))
 				foundMetrics = append(foundMetrics, m.Name)
-				if !matchVal {
+				if skipValAssert {
 					continue
 				}
 
-				if v, ok := expectedMetrics[m.Name]; ok {
-					if dp, ok := v.(int); ok {
-						assert.Equal(t, int64(dp), d.DataPoints[0].Value, m.Name)
-					} else {
-						assert.Fail(t, "expected an int value", m.Name)
-					}
-				} else if match {
+				if fn, ok := expectedMetrics[m.Name]; ok {
+					assert.True(t, fn(t, d.DataPoints[0].Value), m.Name)
```

Comment:
wouldn't it assert one time within the AssertionFunc and another time in `assert.True`?